signal2html/html.py

```diff
@@ -300,8 +341,15 @@ def dump_thread(thread: Thread, output_dir: str):
                     }
                 )
 
+        if is_empty_message(out):
+            continue
```

Comment:
Any idea what these corresponds to? Some message types we don't know how to parse?
In any case, perhaps an info message with the message ID (and type) would be appropriate.

Reply:
There are three message types that seem to fall into this category. Here they are matched to the signal message type description (marked (a), (b), (c), with my understanding of the messages):

```java
  47   │    *      _____________________________________ ENCRYPTION ({@link #ENCRYPTION_MASK})
  48   │    *     |        _____________________________ SECURE MESSAGE INFORMATION (no mask, but look at {@link #SECURE_MESSAGE_BIT})
  49   │    *     |       |     ________________________ GROUPS (no mask, but look at {@link #GROUP_UPDATE_BIT})
  50   │    *     |       |    |       _________________ KEY_EXCHANGE ({@link #KEY_EXCHANGE_MASK})
  51   │    *     |       |    |      |       _________  MESSAGE_ATTRIBUTES ({@link #MESSAGE_ATTRIBUTE_MASK})
  52   │    *     |       |    |      |      |     ____  BASE_TYPE ({@link #BASE_TYPE_MASK})
  53   │    *  ___|___   _|   _|   ___|__    |  __|_
  54   │    * |       | |  | |  | |       | | ||    |
  55   │    * 0000 0000 0000 0000 0000 0000 0000 0000
  56 + │    * 0000 0000 1010 0100 0000 0000 0001 0100 (a)    / incoming, expiration timer update, secure push message /
  57 + │    * 0000 0000 1010 0000 0000 0000 0001 0100 (b)    / incoming, secure push message /
  58 + │    * 0000 1000 1010 0000 0000 0000 0001 0100 (c)    / incoming, secure push message, no remote session /
```

After digging into this a bit more it turns out that:

-  ``(a)`` corresponds to ``<contact> disabled disappearing messages.``
-  ``(b)`` looks like an empty message in the app (I have only one of these)
-  ``(c)`` is in the "Note to Self" conversation and reads "message encrypted for non-existing session

I don't necessarily think it's that necessary to support these, what do you think?

Reply:
- I would like to support (a), as well as the equivalent enabling.
- (b) is in fact a real empty message -- then it should also be kept, just as in the app, I think.
- (c) seems like the result of a bug? Probably this can be omitted.

If you don't feel like parsing (a) for now, we can put it as a future enhancement.

BTW, very nice comment! :)